### PR TITLE
voxelized features

### PR DESCRIPTION
**Pull request by evanfeinberg**
in this PR, we introduce a more generalized mechanism for choosing to produce a flat vector or a tensor for a given set of features. pylint and autopep8 has also been used to clean up formatting.

### CODE REVIEW

File: vs_utils/utils/grid_featurizer.py [hunk 1]
```diff
@@ -30,41 +40,86 @@ def generate_random_unit_vector():
...
+    a. see function generate_random_unit_vector() for details
     2. Generate a second random unit vector thru the algorithm in (1), output v
-      a. If absolute value of u \dot v > 0.99, repeat. This is important for numerical stability 
-        (intuition: we want them to be as linearly independent as possible or else the 
-          orthogonalized version of v will be much shorter in magnitude compared to u. I assume
-          in Stack they took this from Gram-Schmidt orthogonalization?)
-      b. v' = v - (u \dot v)*u, i.e. subtract out the component of v that's in u's direction 
-      c. normalize v' (this isn't in Stack but I assume it must be done)
-    3. find w = u \cross v' 
+    a. If absolute value of u \dot v > 0.99, repeat. This is important for numerical stability
```
Comment (rbharath) [inline vs_utils/utils/grid_featurizer.py]: I think autopep8 might have messed this part up. The original extra indentation for a, b, c looked nicer.

**Comment by rbharath** [comment]
LGTM. I'm not doing an in-depth review of this code since we're battle-testing it already for our debugging purposes. We can fix the inevitable bugs in following PRs.

Feel free to go ahead and merge once you fix the minor indenting issue.

**Comment by evanfeinberg** [comment]
This PR update includes several new featurization schemes, including hydrogen bonds, sybyl atom types, pi_pi stacking, pi_cation, salt bridges. It adds ability to assign gasteiger partial charges to all atoms, in protein and ligand, and to add hydrogens corrected for pH. The method for voxelization has also been generalized so it is more straightforward to translate a list or dict of a feature type into a voxel grid. Other minor changes have been made to clean the code.

**Comment by rbharath** [comment]
Going to go ahead and merge this code in since it was already LGTM'd.

**Comment by rbharath** [comment]
This should be merged in, but there are a couple merge conflicts. I took the changes in PR, merged in with master and made new PR #96 that was subsequently merged.

**Pull request merged**